src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -0,0 +1,132 @@
+                    employerInfos =  MapEmployerInfo(relations);
+                    await AppendAddressToLegalEntitiesAsync(employerInfos);
+                }
+
+                _logger.LogError("An invalid response received when trying to get provider relationships");
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, "Error trying to retrieve titles.", null);
+            }
+            catch (JsonReaderException ex)
+            {
+                _logger.LogError(ex, $"Couldn't deserialise {nameof(VacancyApiSearchResponse)}.", null);
+            }
+
+            return employerInfos;
+        }
+
+        private async Task AppendAddressToLegalEntitiesAsync(IEnumerable<EmployerInfo> employerInfos)
+        {            
+            foreach(var employerInfo in employerInfos)
+            {
+                var providerLegalEntities = employerInfo.LegalEntities;
+                var employerLegalEntities = await _jobsVacancyClient.GetEmployerLegalEntitiesAsync(employerInfo.Id);
+
+                foreach(var providerLegalEntity in providerLegalEntities)
+                {
+                    var source = employerLegalEntities.FirstOrDefault(e => e.LegalEntityId == providerLegalEntity.LegalEntityId);
+                    if (source != null)
+                    {
+                        providerLegalEntity.Address = source.Address;
+                    }                    
+                }
+            }
+        }
+
+        private IEnumerable<EmployerInfo> MapEmployerInfo(ProviderPermissions relations)
+        {
+            var result = new List<EmployerInfo>();
+
+            return relations
+                .AccountProviderLegalEntities
+                .GroupBy(e => e.AccountPublicHashedId)
+                .Select(e => new EmployerInfo() { 
+                    Id = e.Key, 
+                    Name = e.First().AccountName,
+                    LegalEntities = e.Select(l => new LegalEntity{ LegalEntityId = l.AccountLegalEntityId, Name = l.AccountLegalEntityName }) 
+                });
+        }
+
+        private HttpClient CreateHttpClient(ProviderRelationshipApiConfiguration configuration)
+        {
+            
+            var httpClient = new HttpClientBuilder()
+                .WithDefaultHeaders()
+                .WithBearerAuthorisationHeader(new AzureADBearerTokenGenerator(configuration))
+                .Build();
+            
+            httpClient.BaseAddress = new Uri(_configuration.ApiBaseUrl);
+
+            return httpClient;
+        }
+
+        private string AddQueryString(string uri, object queryData)
+        {
+            var queryDataDictionary = queryData.GetType().GetProperties()
+                .ToDictionary(x => x.Name, x => x.GetValue(queryData)?.ToString() ?? string.Empty);
+            return QueryHelpers.AddQueryString(uri, queryDataDictionary);
+        }
+    }
+
+    class ProviderPermissions
```

Comment:
Could these two classes be put in an internal separate class files within the same namespace.

Reply:
done